### ISSUE TITLE
fails to download and just nothing happens until it restarts

### ISSUE THREAD

**Issue by OneOfOne** [ogri-la/wowman#16]
I have a *lot* of addons installed, after it finishes the initial scan and tries to update, it does nothing.

the full log (I set debug to true in config.json):

```
2019-06-22 04:06:58.775 :info [wowman.core:711] starting app
2019-06-22 04:06:58.840 :info [wowman.ui.gui:633] starting gui
2019-06-22 04:06:59.313 :info [wowman.core:364] loading addon summaries from catalog: /home/oneofone/.local/share/wowman/catalog.json
2019-06-22 04:06:59.862 :info [wowman.core:348] (re)loading installed addons: /games/World of Warcraft/_retail_/Interface/AddOns
2019-06-22 04:06:59.893 :warn [wowman.toc:128] failed to find .toc file: /games/World of Warcraft/_retail_/Interface/AddOns/SlootAuras
2019-06-22 04:07:00.178 :info [wowman.core:456] matching installed addons to online addons
2019-06-22 04:07:00.253 :info [wowman.core:471] num installed 206 , num matched 111
2019-06-22 04:07:00.254 :warn [wowman.core:474] you need to manually search for them and then re-install them
2019-06-22 04:07:00.254 :warn [wowman.core:475] failed to match 95 installed addons to online addons: details!-raid-check-(plugin), raid|cff7fff7fachievement|r---legion-raids, |cff1784d1elvui|r-|cff9482c9shadow-&-light|r, details!-encounter-breakdown-(plugin), raider.io-|cffffffffmythic-plus|r-(taiwan---alliance), raider.io-|cffffffffmythic-plus|r-(taiwan---horde), |cffffe00a<|r|cffff7d0adbm|r|cffffe00a>|r-|cff69ccf0party-bfa|r, bigwigs-[|cffeda55fazeroth|r], details-blackrock-foundry, |cffffe00a<|r|cffff7d0adbm|r|cffffe00a>|r-|cff69ccf0darkmoon-faire|r,raider.io-|cffffffffraiding|r-(taiwan---horde), |cffffe00a<|r|cffff7d0adbm|r|cffffe00a>|r-|cff69ccf0uldir|r, raid|cff7fff7fachievement|r---wotlk-dungeons, details-highmaul, handynotes-battle-for-azeroth-achievments, |cffffe00a<|r|cffff7d0adbm|r|cffffe00a>|r-|cff69ccf0the-eternal-palace|r, maxdps, raid|cff7fff7fachievement|r---wod-raids, raider.io-|cffffffffraiding|r-(americas---horde), stop-reseting-my-camera-please, handynotes_suramartelemancy, raid|cff7fff7fachievement|r---naxxramas, masque-|cffff8000muq|r, raider.io-|cffffffffraiding|r-(europe---alliance), |cff1784d1elvui-|rmicrobar-enhancement, new-|cffffff88openables|r, wim, weakauras-model-paths, raid|cff7fff7fachievement|r---cataclysm-raids, adibags-configuration, |cffffe00a<|r|cffff7d0adbm|r|cffffe00a>|r-|cff0055ffoptions-gui|r, nameplatesct, handynotes_warfrontrares, |cffffe00a<|r|cffff7d0adbm|r|cffffe00a>|r-|cff69ccf0battle-of-dazar'alor|r, raider.io-|cffffffffmythic-plus|r-(americas---horde), maxdps_demonhunter, |cffffe00a<|r|cffff7d0adbm|r|cffffe00a>|r-|cff69ccf0azeroth-(bfa)|r, bigwigs-[|cffeda55fbattle-of-dazar'alor|r], |cffffe00a<|r|cffff7d0adbm|r|cffffe00a>|r-|cff69ccf0world-events|r, bigwigs, bigwigs-[|cffeda55fcrucible-of-storms|r], overachiever---tabs, raid|cff7fff7fachievement|r---wod-dungeons, raider.io-|cffffffffraiding|r-(taiwan---alliance), |cfffe7b2celvui|r, maxdps_druid, weakauras-templates, qe-dungeontips, bigwigs-[|cffeda55fplugins|r], weakauras-companion, wim---elvui-skin, details-target-caller, plater, msbt-[|cffffff80options|r], details!-3d-model-viewer, |cff74d06cre|rporter, details!-tiny-threat-(plugin), |cffffe00a<|r|cffff7d0adbm|r|cffffe00a>|r-|cff69ccf0brawlers|r, raider.io-|cffffffffmythic-plus|r-(korea---horde), |cffffe00a<|r|cffff7d0adbm|r|cffffe00a>|r-|cff0055ffstatus-bar-timers-(required)|r, raid|cff7fff7fachievement|r---ulduar, raider.io-|cffffffffraiding|r-(korea---alliance), raider.io-|cffffffffmythic-plus|r-(europe---horde), rclootcouncil---extra-utilities, details-throne-of-thunder, details-siege-of-orgrimmar, overachiever---cache, raider.io-|cffffffffmythic-plus|r-(korea---alliance), |cff1784d1elvui|r-|cff4beb2ccustomtags|r, bigwigs-[|cffeda55fuldir|r], bigwigs-[|cffeda55foptions|r], raider.io-|cffffffffmythic-plus|r-(europe---alliance), details!-vanguard-(plugin), overachiever---trade, raid|cff7fff7fachievement|r---legion-dungeons, handynotes_legiontownportals, handynotes-tools-of-the-trade, raider.io-|cffffffffmythic-plus|r-(americas---alliance), |cffffe00a<|r|cffff7d0adbm|r|cffffe00a>|r-|cff0055ffdefault-skin-(required)|r, flash-task-bar, raider.io-|cffffffffraiding|r-(europe---horde), macrotoolkit-icon-data, raid|cff7fff7fachievement|r---cataclysm-dungeons, details!-storage, raider.io-|cffffffffraiding|r-(korea---horde), maxdps_monk, rclootcouncil---group-gear, |cff1784d1elvui-|raura-bars-mover, capping_options, raider.io-|cffffffffraiding|r-(americas---alliance), |cfffe7b2celvui-config|r, |cffffe00a<|r|cffff7d0adbm|r|cffffe00a>|r-|cff69ccf0crucible-of-storms|r, raid|cff7fff7fachievement|r---icecrown, details!-streamer-(plugin), weakauras-options
2019-06-22 04:07:00.425 :info [wowman.core:556] checking for updates
2019-06-22 04:07:01.476 :info [wowman.http:120] downloading summary data: brewmastertools
Jun 21, 2019 11:07:02 PM org.apache.http.client.protocol.ResponseProcessCookies processCookies
WARNING: Invalid cookie header: "Set-Cookie: AWSALB=DatOD4vIcrDAUXIOP/Nq3KtXQNpnkHRybofU+EBLspG1IEdXBtY6B4/B6qM3gy28vQn5MNs9X8QXu4eaCv7eJaoev6InVuxrXLo+3WS3wxtKHw7rVdXMib6wNsm7; Expires=Sat, 29 Jun 2019 04:07:01 GMT; Path=/". Invalid 'expires' attribute: Sat, 29 Jun 2019 04:07:01GMT
```

*edit* Thanks for your work on this btw.

**Comment by OneOfOne** [comment]
I deleted that specific addon and it started downloading more data.

**Comment by torkus** [comment]
hi! thanks for making an issue. I'll take a look at this today

> num installed 206

yeah, wow :)

everything is painfully *serial* right now, which means one download will happen after another rather than in parallel. And curseforge is quite a bit slower than wowinterface. *And* there is no progress bar to show how large an addon is or how far along the download is.

**Comment by torkus** [comment]
looks like it's some strange Curseforge behaviour. 

You can see the addon is very definitely available in their catalog here: https://www.curseforge.com/wow/addons/search?search=brewmastertools

it's just the page itself returns a 404 not found:
https://www.curseforge.com/wow/addons/brewmastertools

that said, wowman should definitely not hang or hold up the updates of addons when encountering a 404

**Comment by torkus** [comment]
Not sure if this helps you, but the wow.curseforge has more information:
https://wow.curseforge.com/projects/brewmastertools

and their github page a bit more:
https://github.com/emptyrivers/BrewmasterTools

**Comment by torkus** [comment]
Good news, I can replicate what is happening to you. I'll update this ticket again when I make a point release available with the bug fix.

> edit Thanks for your work on this btw.

my pleasure

**Comment by torkus** [comment]
fixed in 0.7.1: https://github.com/ogri-la/wowman/releases/tag/0.7.1

if you're on Arch, 0.7.1 is also available in the AUR.

**Comment by OneOfOne** [comment]
Awesome, thank you very much! good luck next week in game!
